sdk/kiosk/src/query/kiosk.ts

```diff
@@ -35,21 +37,21 @@ export async function fetchKiosk(
 	const lockedItemIds: string[] = [];
 
 	// extracted kiosk data.
-	const kioskData = extractKioskData(data, listings, lockedItemIds);
+	const kioskData = extractKioskData(data, listings, lockedItemIds, kioskId);
 
 	// split the fetching in two queries as we are most likely passing different options for each kind.
 	// For items, we usually seek the Display.
 	// For listings we usually seek the DF value (price) / exclusivity.
-	const [kiosk, listingObjects] = await Promise.all([
+	const [kiosk, listingObjects, items] = await Promise.all([
 		options.withKioskFields ? getKioskObject(client, kioskId) : Promise.resolve(undefined),
 		options.withListingPrices
-			? client.multiGetObjects({
-					ids: kioskData.listingIds,
-					options: {
-						showContent: true,
-					},
+			? getAllObjects(client, kioskData.listingIds, {
+					showContent: true,
 			  })
 			: Promise.resolve([]),
+		options.withObjects
+			? getAllObjects(client, kioskData.itemIds, options.objectOptions || { showDisplay: true })
```

Comment:
Ah, true. Much better! 👍

Reply:
No strong opinions here, but we can either do `{ showDisplay: true,  ...options.objectOptions }` which would allow someone to pass `{ showDisplay: false }` to overwrite this option, or `{ ...options.objectOptions, showDisplay: true }`. 

I didn't trace this through too far, but I think passing other object options shouldn't change the default showDisplay option.   If we depend on the display, we should force it to true, otherwise we can let it be overwritten with an explicitly `showDisplay: false`

Reply:
We don't really have any dependency here (the SDK doesn't rely on that data). It's mostly a convenient way to fetch all items (with multi-pages support), instead of having the users do it on their own. 

That's why I kept it with either the one or the other (the SDK, by default, fetches the display, but someone could choose to fetch with a different setup, like they are using `multiGetObjects`).

The idea of allowing someone to pass `showDisplay: false` feels a bit unusual, since I don't think anyone is passing negative flags (on the main SDK too), they just omit them.